core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -147,9 +148,12 @@ public InternalEngine(EngineConfig engineConfig) throws EngineException {
         if (engineConfig.isAutoGeneratedIDsOptimizationEnabled() == false) {
             maxUnsafeAutoIdTimestamp.set(Long.MAX_VALUE);
         }
-        deletionPolicy = new SnapshotDeletionPolicy(new KeepOnlyLastCommitDeletionPolicy());
         this.uidField = engineConfig.getIndexSettings().isSingleType() ? IdFieldMapper.NAME : UidFieldMapper.NAME;
         this.versionMap = new LiveVersionMap();
+        final TranslogDeletionPolicy translogDeletionPolicy = new TranslogDeletionPolicy();
+        this.deletionPolicy = new CombinedDeletionPolicy(
```

Comment:
sadly it doesn't fit in one line, I'll make it like this - it seems you preft it:

```
        this.deletionPolicy = new CombinedDeletionPolicy(
            new SnapshotDeletionPolicy(new KeepOnlyLastCommitDeletionPolicy()), translogDeletionPolicy, openMode);
```